commit 738e1e694bf6d9e965ba0ce2667a1bec5780a172
Author: Petr Vandrovec <petr@vandrovec.name>
Date:   Sun Oct 12 20:58:29 2008 -0700

    r8169: NULL pointer dereference on r8169 load
    
    mmio_addr in r8169 needs to be initialized before use
    
    Maybe that all tp-> initialization should be moved before rtl_init_mac_address call,
    but this is enough to get rid of crash in rtl_rar_set due to mmio_addr being uninitialized.
    
    Signed-off-by: Petr Vandrovec <petr@vandrovec.name>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index bdae2c5..c821da2 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -2154,6 +2154,8 @@ rtl8169_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 
 	spin_lock_init(&tp->lock);
 
+	tp->mmio_addr = ioaddr;
+
 	rtl_init_mac_address(tp, ioaddr);
 
 	/* Get MAC address */
@@ -2186,7 +2188,6 @@ rtl8169_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 #endif
 
 	tp->intr_mask = 0xffff;
-	tp->mmio_addr = ioaddr;
 	tp->align = cfg->align;
 	tp->hw_start = cfg->hw_start;
 	tp->intr_event = cfg->intr_event;
